oss_src/sframe/rolling_aggregate.cpp

```diff
@@ -0,0 +1,178 @@
+#include <algorithm>
+#include <flexible_type/flexible_type.hpp>
+#include <boost/circular_buffer.hpp>
+#include <boost/algorithm/string.hpp>
+#include <sframe/rolling_aggregate.hpp>
+
+namespace graphlab {
+namespace rolling_aggregate {
+
+ssize_t clip(ssize_t val, ssize_t lower, ssize_t upper) {
+  return std::min(upper, std::max(lower, val));
+}
+
+// Calculate the size of an inclusive range
+size_t calculate_window_size(ssize_t start, ssize_t end) {
+  if((start < 0) && (end >= 0)) {
+    return size_t(std::abs(start) + end + 1);
+  }
+
+  auto the_size = size_t(std::abs(end - start) + 1);
```

Comment:
Just curious, is there a reason you're using "auto" on the left hand side, rather than size_t?